### PR TITLE
feat: read state in batches

### PR DESCRIPTION
**Pull request by ppedziwiatr**
+ AbortController, CC @dtfiedler ....

### CODE REVIEW

File: src/contract/HandlerBasedContract.ts [hunk 1]
```diff
@@ -200,27 +210,83 @@ export class HandlerBasedContract<State> implements Contract<State> {
...
+      }
+      this.logger.debug(`Evaluating ${interactions.length} in ${batchesLoaded}`);
+      cachedState = await this.readStateFor(cachedState?.sortKey || genesisSortKey, interactions);
+      if (signal.aborted) {
+        throw new Error('readStateBatch aborted');
+      }
+      this.logger.debug(
+        `Batch ${batchesLoaded} evaluated in ${batchBenchmark.elapsed()} at sortKey ${cachedState.sortKey}`
+      );
+    } while (interactions.length > 0);
```
[the remark this answers is not in the document]
Comment (ppedziwiatr) [reply src/contract/HandlerBasedContract.ts:258]: good catch, though the issue was in the createExecutionContext, not this loop itself - should be fixed now.

File: src/contract/HandlerBasedContract.ts [hunk 2]
```diff
@@ -200,27 +209,86 @@ export class HandlerBasedContract<State> implements Contract<State> {
...
+      input,
+      'view',
+      caller,
+      undefined,
+      tags,
+      transfer,
+      false,
+      false,
+      true,
+      signal
```
Comment (Tadeuchi) [inline src/contract/HandlerBasedContract.ts:281]: We should consider some callContractOptions with default values. 
Not necessarly, but maybe in the future?

File: src/core/modules/impl/ArweaveGQLTxsFetcher.ts [hunk 3]
```diff
@@ -138,10 +137,26 @@ export class ArweaveGQLTxsFetcher {
...
+  ): Promise<GQLEdgeInterface[]> {
     let pageResult = (await this.fetch<GQLResultInterface['data']>(TRANSACTIONS_QUERY, variables)).transactions;
     const edges: GQLEdgeInterface[] = [...pageResult.edges];
-    while (pageResult.pageInfo.hasNextPage) {
+    let pagesLoaded = 1;
+    if (pagesLoaded >= pagesPerBatch) {
+      return edges;
+    }
+    if (signal?.aborted) {
+      throw new Error(`Abort signal in ${ArweaveGQLTxsFetcher.name}`);
```
Comment (Tadeuchi) [inline src/core/modules/impl/ArweaveGQLTxsFetcher.ts:152]: Maybe this error should be of type `AbortError`?
similar to undici
https://github.com/nodejs/undici/blob/3fd7bf3a34f593c7d4fb03e48559755c659d1f1a/lib/fetch/index.js#L108C25-L108C25

File: src/contract/HandlerBasedContract.ts (hunk 2 above)
Comment (ppedziwiatr) [reply src/contract/HandlerBasedContract.ts:281]: yeah, in general w should refactor to object params in most of the Contract interface methhod.

File: src/core/modules/impl/ArweaveGQLTxsFetcher.ts (hunk 3 above)
Comment (ppedziwiatr) [reply src/core/modules/impl/ArweaveGQLTxsFetcher.ts]: a dedicated AbortError has been added.

**Pull request merged**